File: ios/AITutor/Domain/Repository/GroupChatRepository.swift
import Foundation

protocol GroupChatRepository: AnyObject {
    func listRooms() async throws -> [Room]
    func createRoom(name: String) async throws -> Room
    func joinRoom(roomId: String) async throws

    /// Async stream of message snapshots for a room. Cancelling the iterator
    /// closes the realtime channel.
    func observeMessages(roomId: String) -> AsyncStream<[GroupMessage]>

    func sendMessage(roomId: String, content: String) async throws
}

File: android/app/src/main/java/com/aitutor/app/domain/repository/GroupChatRepository.kt
package com.aitutor.app.domain.repository

import com.aitutor.app.domain.model.GroupMessage
import com.aitutor.app.domain.model.Room
import kotlinx.coroutines.flow.Flow

interface GroupChatRepository {
    suspend fun listRooms(): Result<List<Room>>
    suspend fun createRoom(name: String): Result<Room>
    suspend fun joinRoom(roomId: String): Result<Unit>

    /** Cold flow — calling collect() opens the realtime channel; cancelling closes it. */
    fun observeMessages(roomId: String): Flow<List<GroupMessage>>

    suspend fun sendMessage(roomId: String, content: String): Result<Unit>
}
